### PR TITLE
howto: add vm template introduction and guide

### PR DESCRIPTION
**Pull request by bergwolf**
So that we have a doc to point to when people asks about VM templating.

Fixes: #347

### CODE REVIEW

File: how-to/what-is-vm-templating-and-how-do-I-use-it.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# What Is VM Template and How To Enable It
+
+### What is VM template
+VM template is a Kata containers feature that enables new VMs
```
[the remark this answers is not in the document]
Comment (grahamwhaley) [reply how-to/what-is-vm-templating-and-how-do-I-use-it.md]: s/containers/Containers/

File: how-to/what-is-vm-templating-and-how-do-I-use-it.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+here](https://github.com/kata-containers/runtime/pull/303#issuecomment-395846767).
+
+In another example, we creates 10 Kata containers with containerd shimv2
+and calculated the average boot up speed for each of them. And the result
+showed that VM template speeds up Kata containers creation by as much as
+38.68%. See [full results here](https://gist.github.com/bergwolf/06974a3c5981494a40e2c408681c085d).
+
+### What are the Cons
+One drawback of VM templating is that it cannot avoid cross-VM side-channel
+attack such as [CVE-2015-2877](https://access.redhat.com/security/cve/cve-2015-2877)
```
Comment (grahamwhaley) [inline how-to/what-is-vm-templating-and-how-do-I-use-it.md]: I'm OK pointing to RH for the CVE - or we can point to one of the core sources such as https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2015-2877 - which are referenced off the RH item, so not a biggy.

**Comment by bergwolf** [comment]
Thanks for reviewing @grahamwhaley @jodh-intel . PR updated.

**Comment by jodh-intel** [comment]
Nice!

lgtm

ping @klynnrif.

**Comment by liwei** [comment]
@bergwolf nice work! I have setup vm templating support according to this document.
one small sugguestion, may be we should mention the dependency of qemu-lite which took my some time to struggle vm templating failure because I have used qemu-vanilla.

**Pull request merged**